### PR TITLE
atuin: init at 0.7.1

### PR DESCRIPTION
**Pull request by onsails**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/atuin/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+with rustPlatform;
+buildRustPackage rec {
+  pname = "atuin";
+  version = "0.6.4";
+
+  src = fetchFromGitHub {
+    owner = "ellie";
+    repo = "atuin";
```
Comment (fabaff) [inline pkgs/tools/misc/atuin/default.nix:10]: Opinionated change.

```suggestion
    repo = pname;
```

File: pkgs/tools/misc/atuin/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+with rustPlatform;
+buildRustPackage rec {
```
Comment (fabaff) [inline pkgs/tools/misc/atuin/default.nix:4]: One input per line makes reviews in the future easier.

```suggestion
{ lib
, rustPlatform
, fetchFromGitHub
}:

rustPlatform.buildRustPackage  rec {
```

File: pkgs/tools/misc/atuin/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    owner = "ellie";
+    repo = "atuin";
+    rev = "v${version}";
+    sha256 = "sha256-JmSnOQJ2rQQZaenn8FWM5LYtSDsOkBGloX8PoX9nT70=";
+  };
+
+  cargoSha256 = "0j65njw67737w4f3l8d8yiq370qlczrjhnli3z1kfsjpdh5yhw11";
+
+  meta = with lib; {
+    description = "Magical shell history";
```
Comment (fabaff) [inline pkgs/tools/misc/atuin/default.nix:18]: Not much magic here :wink: " database" and the possibility to record "additional context" seems to be features.

Comment (onsails) [reply pkgs/tools/misc/atuin/default.nix:21]: I've just copied the official description :)

File: pkgs/tools/misc/atuin/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+
+with rustPlatform;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/atuin/default.nix:6]:
```suggestion
```

please use the smallest scope possible.

File: pkgs/tools/misc/atuin/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    owner = "ellie";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-JmSnOQJ2rQQZaenn8FWM5LYtSDsOkBGloX8PoX9nT70=";
+  };
+
+  cargoSha256 = "0j65njw67737w4f3l8d8yiq370qlczrjhnli3z1kfsjpdh5yhw11";
+
+  meta = with lib; {
+    description = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/atuin/default.nix:21]:
```suggestion
    description = "...";
    longDescription = ''
```

That to long for description.

**Pull request merged**